### PR TITLE
Add a cost allocation tag to postgres instances

### PR DESCRIPTION
**Pull request by stafot**
#### Summary
If this commit applied the postgres instances will have a tag that is used for cost_allocation

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-36254

#### Release Note

```release-note
Add `multitenant-rds` cost allocation tag to postgres multi-tenant instances.
```

### CODE REVIEW

**Review by spirosoik — APPROVED** [review]

File: terraform/aws/modules/rds-aurora-postgresql/main.tf [hunk 1]
```diff
@@ -73,22 +73,28 @@ resource "aws_rds_cluster" "provisioning_rds_cluster" {
...
+  apply_immediately            = var.apply_immediately
+  monitoring_role_arn          = data.aws_iam_role.enhanced_monitoring.arn
+  monitoring_interval          = var.monitoring_interval
+  promotion_tier               = count.index + 1
+  performance_insights_enabled = local.performance_insights_enabled
 
-  tags = var.tags
+  tags = merge(
+    {
+      "DatabaseType"                        = "multitenant-rds",
```
Comment (stylianosrigas) [inline terraform/aws/modules/rds-aurora-postgresql/main.tf:93]: @stafot please change this DatabaseType tag to be the same as the RDS cluster. It gets the value from a variable now.

**Review by stylianosrigas — CHANGES_REQUESTED** [review]

File: terraform/aws/modules/rds-aurora-postgresql/main.tf (hunk 1 above)
Comment (stafot) [reply terraform/aws/modules/rds-aurora-postgresql/main.tf:93]: ack. And has various values?

**Review by stylianosrigas — APPROVED** [review]
LGTM 👌

File: terraform/aws/modules/rds-aurora-postgresql/main.tf (hunk 1 above)
Comment (stylianosrigas) [reply terraform/aws/modules/rds-aurora-postgresql/main.tf]: yeap `multitenant-rds` and `multitenant-rds-dbproxy`

Comment (stafot) [reply terraform/aws/modules/rds-aurora-postgresql/main.tf]: Hmm, thanks for the info. Need to adjust cost allocation scripts and metrics a bit. 🔥

**Pull request merged**